packages/data-context/src/DataContext.ts

```diff
@@ -398,7 +398,14 @@ export class DataContext {
       }
 
       this.update((d) => {
-        d.diagnostics.warnings.push(warning)
+        const warningsIndex = d.diagnostics.warnings.findIndex((warning) => warning.cypressError.type === err.type)
+
+        if (warningsIndex !== -1) {
+          // older warning is superceeded by newer one (eg - warning same, but the details of the stack trace could be different)
+          d.diagnostics.warnings.splice(warningsIndex, 1)
```

Comment:
Is this logic need to be merged with the [issuedWarnings ](https://github.com/cypress-io/cypress/blob/develop/packages/config/src/browser.ts#L74)logic in `@packages/config`?

Reply:
Done, see below comment/discussion thread.